src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -189,33 +189,28 @@ public function set($id, $service)
         $id = $this->normalizeId($id);
 
         if ('service_container' === $id) {
-            throw new InvalidArgumentException('You cannot set service "service_container".');
+            throw new InvalidArgumentException('You cannot set the service "service_container".');
+        }
+
+        if (isset($this->privates[$id])) {
+            throw new InvalidArgumentException(sprintf('You cannot set the private service "%s".', $id));
```

Comment:
not sure if this should throw actually? (as for below) compared to setting a new service (`Container::$synthetics`?)

thats a bit more work though.. but we have to decide in this PR i guess :)